common/raw_hashtable.h

```diff
@@ -827,6 +841,84 @@ auto BaseImpl<InputKeyT, InputValueT, InputKeyContextT>::InsertImpl(
                     "or an empty slot.";
 }
 
+template <typename InputKeyT, typename InputValueT, typename InputKeyContextT>
+[[clang::noinline]] auto
+BaseImpl<InputKeyT, InputValueT, InputKeyContextT>::GrowImpl(
+    ssize_t target_alloc_size, KeyContextT key_context) -> void {
+  CARBON_CHECK(llvm::isPowerOf2_64(target_alloc_size));
+  if (target_alloc_size < alloc_size()) {
```

Comment:
```suggestion
  if (target_alloc_size <= alloc_size()) {
```
Or is the intent that you can call `Grow(alloc_size())` in order to clear out the tombstones? (If so, then please add that to the documented contract.)

Reply:
No, your suggested change is correct. The code path below won't actually work correctly for clearing out tombstones.

If we want to support that, I'd use a different API and implementation. So not part of this PR.